velox/docs/functions/spark/string.rst

```diff
@@ -427,3 +427,21 @@ String Functions
     Returns string with all characters changed to uppercase. ::
 
         SELECT upper('SparkSql'); -- SPARKSQL
+
+.. spark:function:: varchar_type_write_side_check(string, limit) -> varchar
+
+    Removes trailing space characters (ASCII 32) that exceed the length ``limit`` from the end of input ``string``. ``limit`` is the maximum length of characters that can be allowed.
+    Throws exception when ``string`` still exceeds ``limit`` after trimming trailing spaces or when ``limit`` is negative.
+    When ``limit=0``, this function returns an empty string if the input ``string`` is empty or contains only spaces, but throws an exception for any other input.
+    Spark SQL itself rejects ``VARCHAR(0)`` in DDL statements so this case will not be encountered.
```

Comment:
I wonder how the behavior is determined in Velox's implementation given this.

Reply:
Would it make sense to throw user error for empty input string as Spark SQL also rejects it?

> Spark SQL itself rejects ``VARCHAR(0)`` in DDL statements so this case will not be encountered.

And perhaps then replace this with `Throws for empty input string`.

Reply:
> Empty input string "" is valid input.

Found in the below test that empty result is returned for empty input string, is this expected?

https://github.com/facebookincubator/velox/pull/13034/files#diff-9def44c0e6c971d9c56938870f8a497b79ae76cb6bb199d4780e71f0dacd09cbR40